lib/sprockets/loader.rb

```diff
@@ -196,17 +204,39 @@ def store_asset(asset, unloaded)
         if cached_asset[:metadata]
           # Deep dup to avoid modifying `asset`
           cached_asset[:metadata] = cached_asset[:metadata].dup
-          if cached_asset[:metadata][:included]
+          if cached_asset[:metadata][:included] && !cached_asset[:metadata][:included].empty?
             cached_asset[:metadata][:included] = cached_asset[:metadata][:included].dup
-            cached_asset[:metadata][:included] = cached_asset[:metadata][:included].map {|uri| compress_from_root(uri) }
+            cached_asset[:metadata][:included].map! {|uri| compress_from_root(uri) }
+          end
+
+          if cached_asset[:metadata][:links] && !cached_asset[:metadata][:links].empty?
+            cached_asset[:metadata][:links] = cached_asset[:metadata][:links].dup
+            cached_asset[:metadata][:links].map! {|uri| compress_from_root(uri) }
+          end
+
+          if cached_asset[:metadata][:stubbed] && !cached_asset[:metadata][:stubbed].empty?
+            cached_asset[:metadata][:stubbed] = cached_asset[:metadata][:stubbed].dup
+            cached_asset[:metadata][:stubbed].map! {|uri| compress_from_root(uri) }
+          end
+
+          if cached_asset[:metadata][:required] && !cached_asset[:metadata][:required].empty?
+            cached_asset[:metadata][:required] = cached_asset[:metadata][:required].dup
+            cached_asset[:metadata][:required].map! {|uri| compress_from_root(uri) }
```

Comment:
Minor nitpick, you're missing the space before these:
`cached_asset[:metadata][:required].map! { |uri| compress_from_root(uri) }`